src/pe/certificate_table.rs

```diff
@@ -0,0 +1,152 @@
+/// Implements parsing of pe32's Attribute Certificate Table
+/// See reference:
+/// https://learn.microsoft.com/en-us/windows/win32/debug/pe-format#the-attribute-certificate-table-image-only
+/// https://learn.microsoft.com/en-us/windows/win32/api/wintrust/ns-wintrust-win_certificate
+use crate::error;
+use scroll::Pread;
+
+use alloc::string::ToString;
+use alloc::vec::Vec;
+
+#[repr(u16)]
+#[derive(Debug, PartialEq, Copy, Clone)]
+pub enum AttributeCertificateRevision {
+    /// WIN_CERT_REVISION_1_0
+    Revision1_0 = 0x0100,
+    /// WIN_CERT_REVISION_2_0
+    Revision2_0 = 0x0200,
+}
+
+impl TryFrom<u16> for AttributeCertificateRevision {
+    type Error = error::Error;
+
+    fn try_from(value: u16) -> Result<Self, Self::Error> {
+        Ok(match value {
+            x if x == AttributeCertificateRevision::Revision1_0 as u16 => {
+                AttributeCertificateRevision::Revision1_0
+            }
+            x if x == AttributeCertificateRevision::Revision2_0 as u16 => {
+                AttributeCertificateRevision::Revision2_0
+            }
+            _ => unimplemented!(),
```

Comment:
one last thing, this should not be `unimplemented` but `error::Error`

Reply:
actually it was like this before but @baloo argued we could encounter binaries with unrecognized revisions

Reply:
back to the error

Reply:
oh that's ok if they have unrecognized versions, but `unimplemented` macro is used to put "holes" in branches/variables/etc., and will panic when encountered, so that you can go back later and implement this, but it shouldn't be in any public facing library/"production" code

Reply:
ha my bad. Yeah I meant that the error message should convey the revision was not supported, not that we should panic :(